packages/rpc/json/error.ts

```diff
@@ -0,0 +1,64 @@
+import type { JsonRpcResponse } from "./provider"
+
+// https://eips.ethereum.org/EIPS/eip-1193#errors
+// ProviderRpcError need to implement code, message, data
+
+export class ProviderRpcError extends Error {
+  public readonly code: number
+  public readonly message: string
+  public readonly data?: unknown
+  public readonly jsonrpc?: string
+  public readonly id?: number
```

Comment:
nit: I realized that we might not need these two fields 🙈

Reply:
[remove jsonrpc and id from ProviderRpcError](https://github.com/pilagod/waallet-extension/pull/179/commits/bf011b9081961a35315da2c02d033c46c9401508)